integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/search/ElasticsearchKnnPredicateSpecificsIT.java

```diff
@@ -183,29 +184,64 @@ void knnPredicateInWrongPlace_addingPrebuiltKnn() {
 		StubMappingScope scope = index.createScope();
 
 		// all good:
-		SearchPredicate knn = scope.predicate().knn( 15 ).field( "location" ).matching( 50.0f, 50.0f ).toPredicate();
-		// we add multiple clauses to prevent "optimizations" leading to bool predicate being replaced by a simple knn predicate
-		SearchPredicate boolKnnInShould = scope.predicate().bool().should( knn )
-				.should( scope.predicate().match().field( "parking" ).matching( Boolean.TRUE ) ).toPredicate();
-
-		// adding knn to non-should bool clauses is not ok,
-		//  we only allow knn as a should clause!
-		knnPredicateInWrongPlace( () -> scope.predicate().bool().must( knn ) );
-		knnPredicateInWrongPlace( () -> scope.predicate().bool().mustNot( knn ) );
-		knnPredicateInWrongPlace( () -> scope.predicate().bool().filter( knn ) );
-
-		// adding boolean predicate with a should knn clause as any boolean clause (nesting a correct bool into another one)
-		//  is not ok !
-		knnPredicateInWrongPlace( () -> scope.predicate().bool().should( boolKnnInShould ) );
-		knnPredicateInWrongPlace( () -> scope.predicate().bool().must( boolKnnInShould ) );
-		knnPredicateInWrongPlace( () -> scope.predicate().bool().mustNot( boolKnnInShould ) );
-		knnPredicateInWrongPlace( () -> scope.predicate().bool().filter( boolKnnInShould ) );
-
-		// adding as a knn filter:
-		knnPredicateInWrongPlace(
-				() -> scope.predicate().knn( 10 ).field( "location" ).matching( 50.0f, 50.0f ).filter( knn ) );
-		knnPredicateInWrongPlace(
-				() -> scope.predicate().knn( 10 ).field( "location" ).matching( 50.0f, 50.0f ).filter( boolKnnInShould ) );
+		SearchPredicate inlineKnn = scope.predicate().knn( 15 ).field( "location" ).matching( 50.0f, 50.0f ).toPredicate();
```

Comment:
Things weren't going too terrible, but then I thought that we could pass a knn predicate as  a named predicate ... 🙈 
I've pushed what I have done... but maybe you have an idea for something more simple?

Reply:
No worries 😃. I've made it so all these tests pass, but I think you may not like the way it is done 😄. 
BTW speaking of a named predicate, I was wondering if we should just remove the predicate and have the `ElasticsearchNamedPredicate$Builder` just return the `providedPredicate`. But let's discuss that later 😃